src/Microsoft.AspNet.Identity/IdentityResult.cs

```diff
@@ -51,21 +46,21 @@ public static IdentityResult Failed(params IdentityError[] errors)
         }
 
         /// <summary>
-        ///     Log Identity result
+        ///     Return string representation of IdentityResult
         /// </summary>
-        /// <param name="logger"></param>
-        /// <param name="message"></param>
-        public virtual void Log(ILogger logger, string message)
+        /// <returns>"Succedded", if result is suceeded else "Failed:error codes"</returns>
+        public override string ToString()
         {
-            // TODO: Take logging level as a parameter
-            if (Succeeded)
-            {
-                logger.LogInformation(Resources.FormatLogIdentityResultSuccess(message));
-            }
-            else
-            {
-                logger.LogWarning(Resources.FormatLogIdentityResultFailure(message, string.Join(",", Errors.Select(x => x.Code).ToList())));
-            }
+            return Succeeded ? "Succeeded" : string.Format("{0} : {1}", "Failed", string.Join(",", Errors.Select(x => x.Code).ToList()));
+        }
+
+        /// <summary>
+        ///     Get the level to to log this result at
+        /// </summary>
+        /// <returns></returns>
+        public virtual LogLevel GetLogLevel()
+        {
+            return Succeeded ? LogLevel.Verbose : LogLevel.Warning;
```

Comment:
Is there anything we should be logging as information? I think we should log as information when we write changes to the Identity store (I going down from the top of the PR, so maybe I haven't got to it yet?).